tests/lib/managers/terms-of-service-test.js

```diff
@@ -0,0 +1,500 @@
+
+	beforeEach(function() {
+		mockery.enable({
+			warnOnUnregistered: false
+		});
+		// Register Mocks
+		mockery.registerAllowable(MODULE_FILE_PATH, true);
+		// Setup File Under Test
+		TermsOfService = require(MODULE_FILE_PATH);
+		termsOfService = new TermsOfService(boxClientFake);
+	});
+
+	afterEach(function() {
+		sandbox.verifyAndRestore();
+		mockery.deregisterAll();
+		mockery.disable();
+	});
+
+	describe('getTermsOfServicesByID()', function() {
+		it('should make GET request to get terms of service info when called with ID', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service',
+					id: '1234',
+					status: 'enabled',
+					enterprise: {},
+					tos_type: 'external',
+					text: 'TEST',
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.mock(boxClientFake).expects('get').withArgs('/terms_of_services/1234')
+				.returns(Promise.resolve(response));
+			termsOfService.getTermsOfServicesByID(TERMS_OF_SERVICE_ID);
+		});
+
+		it('should call callback with the terms of service info when a 200 response is returned', function(done) {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service',
+					id: '1234',
+					status: 'enabled',
+					enterprise: {},
+					tos_type: 'external',
+					text: 'TEST',
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.stub(boxClientFake, 'get').returns(Promise.resolve(response));
+			termsOfService.getTermsOfServicesByID(TERMS_OF_SERVICE_ID, function(err, termsOfServiceObject) {
+				assert.ifError(err);
+				assert.strictEqual(termsOfServiceObject, response.body, 'terms of service object is returned');
+			});
+			done();
+		});
+
+		it('should return a promise resolving to a terms of service object when a 200 response is returned', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					type: 'terms_of_service',
+					id: '1234',
+					status: 'enabled',
+					enterprise: {},
+					tos_type: 'external',
+					text: 'TEST',
+					created_at: '2017-10-21T20:09:59-07:00',
+					modified_at: '2017-10-21T20:09:59-07:00'
+				}
+			};
+			sandbox.stub(boxClientFake, 'wrapWithDefaultHandler').returnsArg(0);
+			sandbox.stub(boxClientFake, 'get').returns(Promise.resolve(response));
+			return termsOfService.getTermsOfServicesByID(TERMS_OF_SERVICE_ID)
+				.then(termsOfServiceObject => {
+					assert.strictEqual(termsOfServiceObject, response, 'terms of service object is returned');
+				});
+		});
+	});
+
+
+	describe('getTermsOfServices()', function() {
+		it('should make GET request to get terms of service info when called', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					total_count: 2,
```

Comment:
I don't think it matters, but this `total_count` doesn't match the array below, which might be confusing for someone coming into this file later